basyx.examples.timerseries/src/main/java/org/eclipse/basyx/examples/timeseries/TimeSeriesSubmodel/TimeSeriesSubmodel.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (C) 2023 the Eclipse BaSyx Authors
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+
+package org.eclipse.basyx.examples.timeseries.TimeSeriesSubmodel;
+
+import java.util.function.Function;
+
+import org.eclipse.basyx.aas.bundle.AASBundle;
+import org.eclipse.basyx.submodel.metamodel.api.ISubmodel;
+import org.eclipse.basyx.submodel.metamodel.map.Submodel;
+import org.eclipse.basyx.submodel.metamodel.map.submodelelement.SubmodelElementCollection;
+import org.eclipse.basyx.submodel.metamodel.map.submodelelement.dataelement.MultiLanguageProperty;
+import org.eclipse.basyx.submodel.metamodel.map.submodelelement.operation.Operation;
+
+import com.influxdb.annotations.Measurement;
+
+import org.eclipse.basyx.examples.timeseries.AASXImporter;
+
+/*
+ * Creates ReadRecords operation for a TimeSeries Submodel that follows the IDTA standard
+ * (https://github.com/admin-shell-io/submodel-templates). The operation takes a Range SubmodelElement as input, which
+ * is timerange in ISO UTC format. The output is a SubmodelElementCollection consisting of records
+ * (https://admin-shell.io/idta/TimeSeries/Record/1/1) as found under TimeSeries->Metadata->Record.
+ * Creates another experimental operation ReadRecordsShort which has a similar input as ReadRecords, but
+ * returns the query results as a string in a value of a property (reduces overhead).
+ * 
+ * author Al-Obaidi
+ */
+public class TimeSeriesSubmodel {
```

Comment:
Shouldn't this TimeSeriesSubmodel extend the Submodel?
It is defined in the TimeSeries Submodel Specification.

![image](https://github.com/eclipse-basyx/basyx-java-examples/assets/62088117/ffcb70e5-0c19-4217-bf8d-cc4189c99a87)

Reply:
I think this class should be more understood as a factory (cf. instantiateTimeSeriesSubmodel). Thus, it should follow the factory pattern (*Factory as name, create() instead of instantiateTimeSeriesSubmodel).